docs/Design/Loops/For Each Key.md

```diff
@@ -0,0 +1,182 @@
+# For Each Key
+
+A loop that itterates over a dictionary by key.
+
+## Commands
+
+### `for each key start`
+
+`for each key start : container key string`
+
+Starts a for each key block.
+The first parameter is the container to be itterated over.
+The second parameter is the itterator variable.
+The third parameter is the type of the itterator variable.
+
+## Usage
+```
+for each key start : container key string
+    comment line : ...
+for each end
+```
+
+### CSharp
+```csharp
+foreach (string key in container.Keys)
+{
+    // ...
+}
+```
+
+### Java
+```java
+for (string key : container.keySet()) {
+    // ...
+}
+```
+
+### Python
+```python
+for key in container:
+    # ...
+```
+
+### Ruby
+```ruby
+container.each_key { |key|
+    # ...
+}
+```
+
+### TypeScript
+```typescript
+for (let key: string in container) {
```

Comment:
TypeScript doesn't let you declare a type annotation for this. 

http://www.typescriptlang.org/play/#src=var%20container%20%3D%20%7B%7D%3B%0D%0A%0D%0Afor%20(let%20key%3A%20string%20in%20container)%20%7B%0D%0A%20%20%20%20%0D%0A%7D